### PR TITLE
python3Packages.radio_beam: update to the latest upstream and fix depends

### PR DESCRIPTION
**Pull request by smaret**
###### Description of changes

Update python3Packages.radio_beam to the latest upstream, and update/fix dependencies on the way.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by smaret** [comment]
Result of `nixpkgs-review pr 193223` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>python310Packages.aplpy</li>
    <li>python310Packages.asdf</li>
    <li>python310Packages.pyregion</li>
    <li>python310Packages.spectral-cube</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.spectral-cube</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python310Packages.astroquery</li>
    <li>python310Packages.pytest-astropy</li>
    <li>python310Packages.pytest-astropy-header</li>
    <li>python310Packages.pyvo</li>
    <li>python310Packages.radio_beam</li>
    <li>python310Packages.reproject</li>
    <li>python39Packages.astroquery</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.pytest-astropy-header</li>
    <li>python39Packages.pyvo</li>
    <li>python39Packages.radio_beam</li>
    <li>python39Packages.reproject</li>
  </ul>
</details>

File: pkgs/development/python-modules/pytest-astropy-header/default.nix [hunk 1]
```diff
@@ -37,12 +33,7 @@ buildPythonPackage rec {
 
   checkInputs = [
     pytestCheckHook
-    pytest-cov
     numpy
-    scipy
-    h5py
-    scikitimage
-    astropy
```
Comment (risicle) [inline pkgs/development/python-modules/pytest-astropy-header/default.nix:45]: These can be removed from the arguments now.

**Review by risicle — APPROVED** [review]
Packages build & `nixpkgs-review` reveals no new failures, macos 10.15 & nixos x86_64.

File: pkgs/development/python-modules/pytest-astropy-header/default.nix (hunk 1 above)
Comment (smaret) [reply pkgs/development/python-modules/pytest-astropy-header/default.nix:45]: You're right. Fixed with bf012b5.

**Pull request merged**